cmd/agent/app/agent.go

```diff
@@ -153,14 +153,19 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 }
 
 func registerWithControlPlaneAPIServer(controlPlaneRestConfig *restclient.Config, memberClusterName string) error {
+	kubeClient := kubeclientset.NewForConfigOrDie(controlPlaneRestConfig)
 	karmadaClient := karmadaclientset.NewForConfigOrDie(controlPlaneRestConfig)
 
 	clusterObj := &clusterv1alpha1.Cluster{}
 	clusterObj.Name = memberClusterName
 	clusterObj.Spec.SyncMode = clusterv1alpha1.Pull
 
-	_, err := karmadactl.CreateClusterObject(karmadaClient, clusterObj, false)
-	if err != nil {
+	if _, err := karmadactl.CreateNamespace(kubeClient, util.NamespaceClusterLease); err != nil {
+		klog.Errorf("failed to create cluster object. cluster name: %s, error: %v", memberClusterName, err)
+		return err
+	}
```

Comment:
Can we have a util like `CreateNamespaceIfNotExist` at [here](https://github.com/karmada-io/karmada/blob/master/pkg/util/namespace.go)?

A command(karmada-agent) should not depend on another command(`karmadactl join`) I think.

Reply:
It seems that `karmadactl` still use `kubeclient.Interface` rather than `client.Client`. What about cleaning up in next PR and sharing the same util code in `karmada-agent` and `karmadactl`. In that way, `gclient` can be used.

https://github.com/karmada-io/karmada/blob/2a8dd53604f9fbfaf95d65b34eba5c7728e49c0f/pkg/karmadactl/join.go#L168-L169

Reply:
What we are going to do is creating a namespace here, I didn't mean to do cleanup in this PR.
Just make sure this PR `clean` would be fine :)